llarp/service/endpoint.cpp

```diff
@@ -460,9 +460,33 @@ namespace llarp
     bool
     Endpoint::PublishIntroSet(const EncryptedIntroSet& i, AbstractRouter* r)
     {
-      // publish via near router
-      const auto path = GetEstablishedPathClosestTo(i.derivedSigningKey);
-      return path && PublishIntroSetVia(i, r, path);
+      /// number of routers to publish to
+      static constexpr size_t PublishRedundancy = 2;
+      // publish to set of paths with unique endpoints
+      path::Path::UniqueEndpointSet_t paths;
+      // try 10 times to pick random paths
+      size_t tries = 10;
+      do
+      {
+        --tries;
+        const auto path = PickRandomEstablishedPath();
+        if(path)
+          paths.emplace(path);
+      } while(tries > 0 and paths.size() < PublishRedundancy);
+
+      if(paths.size() != PublishRedundancy)
+        return false;
```

Comment:
Do we need a log statement here, or would some upstream caller effectively do that for us?

Reply:
it propagates